app/src/main/java/org/fossasia/pslab/fragment/InstrumentsFragment.java

```diff
@@ -109,6 +107,18 @@ private void initiateViews(View view, int rows) {
         listView.setItemAnimator(new DefaultItemAnimator());
         listView.setAdapter(applicationAdapter);
     }
+    
+    @Override
+    public void onConfigurationChanged(Configuration newConfig) {
+        super.onConfigurationChanged(newConfig);
+        getActivity().getSupportFragmentManager().beginTransaction().detach(this).attach(this).commitAllowingStateLoss();
+    }
+
+    @Override
+    public void onDestroyView() {
+        super.onDestroyView();
+        ((PSLabApplication) getActivity().getApplication()).refWatcher.watch(this, InstrumentsFragment.class.getSimpleName());
+    }
```

Comment:
Can we revert all these changes? It's just moving methods from below to up. This will cause unnecessary conflicts.